### PR TITLE
Tutorial build tests

### LINKED ISSUE #134: Creating widgets - Tutorial Tests
Please create tests for the examples in Tutorial 003: Creating widgets

> One of the biggest issues with examples on dojotoolkit.org was that new releases of dojo or new releases of browsers would leave examples broken. Without a way to monitor or identify these issues, we could only hope that a user would report the issue sooner, rather than later.
> 
> For dojo.io, examples should include unit tests and be structured in a way that automatically verifies that the example works with the latest version of dojo, or if it is deprecated any only works with the libraries used in the example.

### PR DESCRIPTION
**Pull request by edhager**
My branch started from https://github.com/maier49/dojo.io/tree/update-tutorials-for-beta2.

The goal of this update is to run through the steps to verify that a user will be successful when they download the tutorial demo code.

The dojo.io Grunt script was modified to do the following:
- Copy the tutorial demo directories into the build directory ("_build").
- Run "npm install" on each tutorial demo directory in the build directory.
- Run "dojo test" on each tutorial demo directory in the build directory.  This builds the tutorials demo code and runs the unit tests in Intern.

Fixes #134

### CODE REVIEW

File: support/grunt/config.ts [hunk 1]
```diff
@@ -7,253 +7,323 @@ export interface WebServerConfig {
 	[ key: string ]: Config;
 }
 
+interface GruntConfig {
+	[ key: string ]: any
```
Comment (maier49) [inline support/grunt/config.ts]: Missing a semicolon

File: support/grunt/config.ts [hunk 2]
```diff
@@ -7,253 +7,323 @@ export interface WebServerConfig {
...
+}
+
+interface TutorialDefinitions {
+	[ key: string ]: string
```
Comment (maier49) [inline support/grunt/config.ts]: Missing a semicolon

File: support/grunt/config.ts [hunk 3]
```diff
@@ -7,253 +7,323 @@ export interface WebServerConfig {
...
 
-export const publish = {
-	'gh-pages': {
-		options: {
-			branch: 'gh-pages',
-			cloneDirectory: '<%= distDirectory %>'
-		}
-	}
-};
+const npmConfigObj = gruntConfig[ 'npm-command' ] = <GruntConfig>{};
```
Comment (maier49) [inline support/grunt/config.ts]: I believe our preferred style is `{} as GruntConfig` vs `<GruntConfig> {}`, though I'm not sure that's captured in the styles guide in `dojo/meta` yet.

**Comment by agubler** [comment]
@edhager is there any reason we can't use matrices in travis to run `dojo build` on all the tutorials?

**Comment by agubler** [comment]
superseded by #167

**Pull request closed without merging**